api/common/v1alpha1/ref.go

```diff
@@ -42,6 +42,10 @@ func (k TargetRefKind) Less(o TargetRefKind) bool {
 	return order[k] < order[o]
 }
 
+func AllTargetRefKinds() []TargetRefKind {
+	return maps.Keys(order)
```

Comment:
SortedKeys may be better here as it's stable?

Reply:
Yes, a sorted slice is definite better. But type of the keys is "TargetRefKind" which is not easy to be sorted here. 
We leave this responsibility to futures caller who care about the order?